src/control/cmd/dmg/storage_query.go

```diff
@@ -121,3 +122,28 @@ func (s *smdQueryCmd) Execute(args []string) error {
 	smdQuery(s.log, s.conns, s.Devices, s.Pools)
 	return nil
 }
+
+// devStateQueryCmd is the struct representing the "storage query devie-state" subcommand
+type devStateQueryCmd struct {
+	logCmd
+	connectedCmd
+	Devuuid string `short:"u" long:"devuuid" description:"Device/Blobstore UUID to query" required:"1"`
+}
+
+// Query the SMD device state of the given device
+func devStateQuery(log logging.Logger, conns client.Connect, uuid string) {
+	if uuid == "" {
+		log.Infof("Device UUID needs to be specified\n")
+		return
+	}
+
+	req := &mgmtpb.DevStateReq{DevUuid: uuid}
+
+	log.Infof("Device State Info:\n%s\n", conns.DevStateQuery(req))
+}
+
+// Execute is run when devStateQueryCmd activates
+func (d *devStateQueryCmd) Execute(args []string) error {
+	devStateQuery(d.log, d.conns, d.Devuuid)
```

Comment:
Also, why make these separate functions rather than just running the logic in Execute()?

Reply:
I was just following coding layout of previous commands, but does make more sense to have it all in Execute() in order to return with error

Reply:
Ah, right. Fair enough. That layout is an artifact of a previous mode which gave daos_shell an interactive CLI. We got rid of that but didn't update all of the existing functions.

Reply:
agreed we don't need to separate functions unless you want to reuse logic between the 2 apps server/shell, which we don't for these commands